### ISSUE TITLE
Resolve Travis CI Nightwatch failures since #32 resolution

### ISSUE THREAD

**Issue by markafitzgerald1** [markafitzgerald1/cribbage-analyst#33]
See https://travis-ci.org/markafitzgerald1/cribbage-analyst/jobs/377124797, for example.

**Comment by markafitzgerald1** [comment]
Issue occurring with Node 8 but not Node 6 or 4 on Travis CI. Able to reproduce the issue - at least once - locally on Node 6(.14.1).

**Comment by markafitzgerald1** [comment]
Issue occurring even with `.waitForElementPresent('.analysis:first-of-type', 30000)` - something else must be amiss.

**Comment by markafitzgerald1** [comment]
Unable to locally reproduce issue with CSS `@import url('https://fonts.googleapis.com/css?family=Arvo');` and `#cardIndices, .card-indices { font-family: 'Arvo', serif; }` commented out. Perhaps this is some issue with Nightwatch / Selenium and CSS font `@import` usage?

**Comment by markafitzgerald1** [comment]
Also unable to locally reproduce issue with `#cardIndices, .card-indices { font-family: 'Arvo', serif; }` present but `@import url('https://fonts.googleapis.com/css?family=Arvo');` absent. Issue likely something to do with Nightwatch / Selenium interaction with loading of font face from Google.

**Comment by markafitzgerald1** [comment]
Looks like PhantomJS < 2 has always had web font issues; v2+ may resolve it: https://stackoverflow.com/questions/23404658/how-can-i-use-google-web-fonts-with-phantomjs. This project currently uses an older `gulp-nightwatch`: `0.2.9`, which depends on and uses PhantomJS as it's Nightwatch browser.

**Comment by markafitzgerald1** [comment]
PhantomJS development was halted 4 months ago: https://github.com/ariya/phantomjs/issues/15344. Might need to switch to another headless browser such as Headless Chrome now or soon.

**Comment by markafitzgerald1** [comment]
`gulp-nightwatch@0.2.9` depends on `nightwatch@0.8.18`. This project has Nightwatch configured to run tests using PhantomJS. Locally that's via a manually installed PhantomJS 2.1.1 (latest - _not_ a 1.x as surmised above) (see https://github.com/markafitzgerald1/cribbage-analyst/issues/44 for missing `README.md` documentations). On Travis CI PhantomJS is pre-installed: https://docs.travis-ci.com/user/reference/precise/#Headless-Browser-Testing-Tools.

According to https://stackoverflow.com/questions/23404658/how-can-i-use-google-web-fonts-with-phantomjs Google Fonts serves up `.woff2` fonts to PhantomJS with its default user agent string but ".woff2 is not supported in PhantomJs 2.x." Work-arounds include either (possibly) using a non-default user agent string with PhantomJS to get PhantomJS-compatible fonts served up or download the fonts to localhost from Google Fonts then serve a PhantomJS compatible font up to the front end.

**Comment by markafitzgerald1** [comment]
To override the default PhantomJS user agent (and get TTF, not WOFF2 served to PhantomJS): https://stackoverflow.com/questions/42026707/how-to-change-give-user-agent-definition-to-phantomjs-from-cmd. Might work to resolve this issue. Alternative is (hopefully scripted as part of gulp pipeline) download of font.

**Comment by markafitzgerald1** [comment]
Apparently Travis CI (and likely Nightwatch) supports headless Firefox out of the box and headless Chrome via Travis CI add-on: https://docs.travis-ci.com/user/gui-and-headless-browsers/#Using-the-Firefox-addon-in-headless-mode.

**Comment by markafitzgerald1** [comment]
Locally working and consistently passing on Node LTS 6 headless Chrome and Firefox tests created. Just needs local verification on Node 4 and 8 before git commit and push to test on Travis CI. (Don't close issue on push - close on passing Travis CI build!)

**Comment by markafitzgerald1** [comment]
Issue resolved in `issue-33-move-e2e-tests-from-phantomjs-to-headless-chrome` branch commit 2c53225dedfa137650f3bf8d363558682011562a - creating PR.

**Comment by markafitzgerald1** [comment]
Travis CI build still broken due to `gulp-gh-pages` failing to `gulp deploy` the site. See https://github.com/shinnn/gulp-gh-pages/issues/116 for issue details - might need to move from `gulp-gh-pages` to `gh-pages`.

Local repro output on `develop` branch: ```[17:15:00] Starting 'deploy'...
C:\Users\Mark\Documents\Code\Git repositories\cribbage-analyst\node_modules\gulp-gh-pages\node_modules\gift\lib\commit.js:145
      ref1 = /^.+? (.*) (\d+) .*$/.exec(line), m = ref1[0], actor = ref1[1], epoch = ref1[2];
                                                       ^

TypeError: Cannot read property '0' of null
    at Function.module.exports.Commit.actor (C:\Users\Mark\Documents\Code\Git repositories\cribbage-analyst\node_modules\gulp-gh-pages\node_modules\gift\lib\commit.js:145:56)
    at Function.module.exports.Commit.parse_commits (C:\Users\Mark\Documents\Code\Git repositories\cribbage-analyst\node_modules\gulp-gh-pages\node_modules\gift\lib\commit.js:111:21)
    at C:\Users\Mark\Documents\Code\Git repositories\cribbage-analyst\node_modules\gulp-gh-pages\node_modules\gift\lib\commit.js:71:39
    at ChildProcess.exithandler (child_process.js:191:7)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at maybeClose (internal/child_process.js:862:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:222:5)```

**Comment by markafitzgerald1** [comment]
Trying a reinstall of `npm` dependencies (`rm -rf node_modules && npm install`) to ensure issue still occurs with dependencies currently brought in by `npm`...

**Comment by markafitzgerald1** [comment]
Issue still occurred after reinstall of dependencies. Try move from `gulp-gh-pages` to `gh-pages` next.

**Comment by markafitzgerald1** [comment]
Candidate move from `gulp-gh-pages` to `gh-pages` not built by Travis CI, likely because branch name ends with `gh-pages`. Fix: whitelist it (and gh-pages if easier - not much harmful) via https://docs.travis-ci.com/user/customizing-the-build/#Safelisting-or-blocklisting-branches.

**Comment by markafitzgerald1** [comment]
Try ```
branches:
    only:
        - /.*/
```
this evening - perhaps Travis CI is not building any branch _ending_ in `gh-pages`?

**Comment by markafitzgerald1** [comment]
Resolved in b7d09541477a643020c229fbf2b022bf70525380.
